src/errors.rs

```diff
@@ -1,64 +1,79 @@
-use crate::tenant_security_client::errors::TenantSecurityError;
+pub use crate::tenant_security_client::errors::{
+    KmsError, SecurityEventError, ServiceError, TenantSecretError, TenantSecurityProxyError,
+};
 use crate::vector::crypto::{
     DecryptError as VectorDecryptError, EncryptError as VectorEncryptError,
 };
 
 /// Errors related to IronCore Alloy SDK
-#[derive(Debug, uniffi::Error, PartialEq, Eq)]
-#[uniffi(flat_error)]
+#[derive(Debug, uniffi::Error, PartialEq, Eq, Clone)]
 pub enum AlloyError {
     /// Error while loading configuration.
-    InvalidConfiguration(String),
+    InvalidConfiguration { msg: String },
     /// Error with key used
-    InvalidKey(String),
+    InvalidKey { msg: String },
     /// Error with user input
-    InvalidInput(String),
+    InvalidInput { msg: String },
     /// Errors while encrypting
-    EncryptError(String),
+    EncryptError { msg: String },
     /// Errors while decrypting
-    DecryptError(String),
+    DecryptError { msg: String },
     /// Error when parsing encryption headers/metadata
-    ProtobufError(String),
-    /// Error with requests to TSC
-    TenantSecurityError(String),
+    ProtobufError { msg: String },
+    /// Error when making a request to the TSP
+    RequestError { msg: String },
+    /// Error converting request data to JSON
+    SerdeJsonError { msg: String },
+    /// Error directly from the TSP
+    TspError {
+        err: TenantSecurityProxyError,
+        http_code: u16,
+        tsp_code: u16,
```

Comment:
Is it not possible for this to also be an enum? Even if it can't be on is too unwieldy, we need a comment explaining what the codes are. We could link to https://javadoc.io/doc/com.ironcorelabs/tenant-security-java/latest/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityErrorCodes.html or something for now, though we should probably make a cross-language page explaining errors on the docs site and link to that longer term.

Reply:
The `err` (now called `error`) is the enum, and the `tsp_code` is just the code. Python could put the code into the enum, but I couldn't find a way to make it similar in Kotlin, so I left them separate.